src/app/clusters/unit-localization-server/unit-localization-server.h

```diff
@@ -0,0 +1,62 @@
+/*
+ *
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app-common/zap-generated/cluster-enums.h>
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app-common/zap-generated/ids/Clusters.h>
+#include <app/AttributeAccessInterface.h>
+#include <app/data-model/List.h>
+#include <cstdint>
+#include <lib/core/CHIPError.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace UnitLocalization {
+
+inline constexpr uint16_t kClusterRevision              = 2;
+inline constexpr uint8_t kMinSupportedLocalizationUnits = 2;
+inline constexpr uint8_t kMaxSupportedLocalizationUnits = 3;
+
+class UnitLocalizationServer : public AttributeAccessInterface
+{
+public:
+    // Register for the UnitLocalization cluster on all endpoints.
+    static UnitLocalizationServer & Instance();
+
+    CHIP_ERROR Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder) override;
+
+    CHIP_ERROR SetSupportedTemperatureUnits(DataModel::List<TempUnitEnum> & units);
+    DataModel::List<TempUnitEnum> GetSupportedTemperatureUnits(void) { return mSupportedTemperatureUnits; }
```

Comment:
Since you want to keep `mSupportedTemperatureUnits`, I think that you should return `const DataModel::List<TempUnitEnum> &`. Otherwise you will return a copy of `span`, but the underlying `mUnitsBuffer` will not be copied. Hence, const ref getter seems to be more appropriate here.

Reply:
Done